src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+                        SyntaxFactory.IdentifierName(BeautifyName(DirectCast(initializer, IParameterReferenceOperation).Parameter.Name))))
+                    Dim anonymousObjectCreation = SyntaxFactory.AnonymousObjectCreationExpression(
+                        SyntaxFactory.ObjectMemberInitializer(SyntaxFactory.SeparatedList(Of FieldInitializerSyntax)(fieldInitializers)))
+                    secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction, anonymousObjectCreation)
+                Else
+                    secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction)
+                End If
+
+                Return CreateInvocationExpression(expression, NameOf(Enumerable.SelectMany), {firstLambda, secondLambda})
+            End Function
+
+            Private Function CreateLambdaExpression(anonymousFunction As IAnonymousFunctionOperation, Optional body As VisualBasicSyntaxNode = Nothing) As LambdaExpressionSyntax
+                If body Is Nothing Then
+                    body = MakeIdentifierNameReplacements(anonymousFunction.Body.Operations.First().Syntax)
+                End If
+                Dim parameters = anonymousFunction.Symbol.Parameters.
+                    Select(Function(p) SyntaxFactory.Parameter(SyntaxFactory.ModifiedIdentifier(BeautifyName(p.Name))))
+                Dim parameterList = SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(parameters))
+                Dim header = SyntaxFactory.LambdaHeader(
+                    SyntaxKind.FunctionLambdaHeader,
+                    attributeLists:=Nothing,
+                    modifiers:=Nothing,
+                    SyntaxFactory.Token(SyntaxKind.FunctionKeyword),
+                    parameterList,
+                    asClause:=Nothing)
+                Return SyntaxFactory.SingleLineFunctionLambdaExpression(header, body)
+            End Function
+
+            Private Function ProcessSelectClause(expression As ExpressionSyntax, selectClause As SelectClauseSyntax) As ExpressionSyntax
+                Dim selectExpression = selectClause.Variables.First().Expression
+                ' Avoid trivial Select(Funciton(x) x)
+                Dim variables = selectClause.Variables
+                Dim variable = variables.First()
+                Dim bodyExpression = variable.Expression
+
+                If bodyExpression.Kind() = SyntaxKind.IdentifierName Then
+                    Dim identifierName = DirectCast(bodyExpression, IdentifierNameSyntax)
+                    Dim identifierNames = GetIdentifierNames(_semanticModel, identifierName, _cancellationToken)
+                    If identifierNames.IsDefault OrElse identifierNames.Length = 1 Then
+                        Return expression
+                    End If
+                End If
+
+                Return CreateInvocationExpression(expression, NameOf(Enumerable.Select), selectClause.Variables.First().Expression)
```

Comment:
>selectClause.Variables.First().Expression [](start = 89, length = 41)

Why are we ignoring all variables, but the first one? Select with multiple variables is going to crate selector that create anonymous type with user expressions inside and user given names for members, we should be very careful to properly translate all this. We can discuss offline how to do this.